API/Data/BankAccountRepository.cs

```diff
@@ -0,0 +1,121 @@
+        {
+            _context = context;
+            _mapper = mapper;
+        }
+
+        public async Task AddBankAccountAsync(BankAccount bankAccount)
+        {
+            await _context.BankAccounts.AddAsync(bankAccount);
+            await _context.SaveChangesAsync();
+        }
+
+        public async Task DeleteBankAccountAsync(BankAccount bankAccount)
+        {
+            _context.Remove(bankAccount);
+            await _context.SaveChangesAsync();
+        }
+
+        public async Task<BankAccount> GetBankAccountAsync(int bankAccountId)
+        {
+            return await _context.BankAccounts
+                .SingleOrDefaultAsync(c => c.Id == bankAccountId);
+        }
+
+        public async Task<IEnumerable<BankAccountDto>> GetBankAccountsAsync(int userId)
+        {
+
+            var query =  await _context.BankAccounts
+                .Where(b => b.AppUserId == userId)
+                .Select(x => new BankAccountDto
+                {
+                 Id = x.Id,
+                 AppUserId = x.AppUserId,
+                 Name = x.Name,
+                 LastActive = x.LastActive,
+                 Operations = x.Operations.Select(c => new OperationDto {
+                    Id = c.Id,
+                    Description = c.Description,
+                    Amount = c.Amount,
+                    Date = c.Date,
+                    Name = c.Name,
+                    CategoryId = c.CategoryId,
+                    BankAccountId = c.BankAccountId,
+                 }).OrderByDescending(o => o.Id).Take(3),
+                 })
+                .ToListAsync();
+
+            return query;
+        }
+
+        public async Task<PagedList<BankAccountDto>> GetPaginatedBankAccountsAsync(UserParams userParams, int appUserId)
+        {
+            //var query = _context.BankAccounts
+            //    .Where(b => b.AppUserId == appUserId);
+
+            //query = userParams.OrderBy switch
+            //{
+            //    "date" => query.OrderByDescending(u => u.LastActive),
+            //    _ => query.OrderByDescending(u => u.Name)
+            //};
+            var query = _context.BankAccounts
+                .Where(b => b.AppUserId == appUserId)
+                .Select(x => new BankAccountDto
+                {
+                    Id = x.Id,
+                    AppUserId = x.AppUserId,
+                    Name = x.Name,
+                    LastActive = x.LastActive,
+                    Operations = x.Operations.Select(c => new OperationDto
+                    {
+                        Id = c.Id,
+                        Description = c.Description,
+                        Amount = c.Amount,
+                        Date = c.Date,
+                        Name = c.Name,
+                        CategoryId = c.CategoryId,
+                        BankAccountId = c.BankAccountId,
+                    }).OrderByDescending(o => o.Id).Take(3),
```

Comment:
3

Reply:
Ogolnie tu masz potworzenie kodu chyba? Masz tu i wyzej to samo query praktycznie. Mozesz to gdzies wydzielic i nie potwarzac kodu: DRY